vtex/cli-plugin-redirects
# Replace graphql client with http POST

#### What is the purpose of this pull request?
Patching the error from the ticket [XTNSNS-737](https://vtex-dev.atlassian.net/browse/XTNSNS-737).

#### What problem is this solving?

The graphql client used here was sending a request to rewriter that was responding with a 500 error with the content:
```
{
    "errors": [
        {
            "message": "Must provide Source. Received: undefined",
            "stack": "Error: Must provide Source. Received: undefined\n    at devAssert (/usr/local/app/node_modules/graphql/jsutils/devAssert.js:12:11)\n    at new Parser (/usr/local/app/node_modules/graphql/language/parser.js:82:67)\n    at Object.parse (/usr/local/app/node_modules/graphql/language/parser.js:35:16)\n    at parseAndValidateQueryToSchema (/usr/local/app/node_modules/@vtex/api/lib/service/worker/runtime/graphql/middlewares/query.js:26:32)\n    at /usr/local/app/node_modules/@vtex/api/lib/service/worker/runtime/graphql/middlewares/query.js:49:16\n    at MultilayeredCache.get (/usr/local/app/node_modules/@vtex/api/lib/caches/MultilayeredCache.js:19:43)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at parseAndValidateQuery (/usr/local/app/node_modules/@vtex/api/lib/service/worker/runtime/graphql/middlewares/query.js:48:23)\n    at /usr/local/app/node_modules/@vtex/api/lib/utils/time.js:79:13\n    at /usr/local/app/node_modules/@vtex/api/lib/utils/cancel.js:9:9\n    at /usr/local/app/node_modules/@vtex/api/lib/utils/time.js:79:13\n    at /usr/local/app/node_modules/@vtex/api/lib/utils/cancel.js:9:9\n    at graphqlError (/usr/local/app/node_modules/@vtex/api/lib/service/worker/runtime/graphql/middlewares/error.js:47:9)\n    at /usr/local/app/node_modules/@vtex/api/lib/utils/time.js:79:13\n    at /usr/local/app/node_modules/@vtex/api/lib/utils/cancel.js:9:9\n    at response (/usr/local/app/node_modules/@vtex/api/lib/service/worker/runtime/graphql/middlewares/response.js:23:5)\n    at /usr/local/app/node_modules/@vtex/api/lib/utils/time.js:79:13\n    at /usr/local/app/node_modules/@vtex/api/lib/utils/cancel.js:9:9\n    at injectGraphqlContext (/usr/local/app/node_modules/@vtex/api/lib/service/worker/runtime/graphql/middlewares/context.js:14:5)\n    at /usr/local/app/node_modules/@vtex/api/lib/utils/time.js:79:13\n    at /usr/local/app/node_modules/@vtex/api/lib/utils/cancel.js:9:9",
            "name": "Error",
            "extensions": {
                "code": "INTERNAL_SERVER_ERROR"
            },
            "forwardedProto": "http",
            "operationId": "c3351971-521e-41e1-9946-56ee07297c00",
            "query": {
                "variables": ""
            },
            "requestId": "9208bdb7c76444a98e9fc61b93068a0a"
        }
    ]
}
```

It gives an idea that the body was not being properly parsed. There is also a big mismatch in versions of the node-vtex-api, the plugin is in the version 3.77.0 while rewriter is at the latest 6.47.0, and there are differences on how body is expected to be parsed. For our applications we always bump the version of the node-vtex-api, but for toolbelt plugins we don't.

There is an important highlight here, toolbelt plugins can't be upgraded to latest node-vtex-api version because we don't offer any way to instantiate clients defining the context, for example in this plugin it makes use of the following createClient method to init a GraphQL Client and I couldn't find a parallel within the latest node-vtex-api version.

```
public static createClient(customContext: Partial<IOContext> = {}, customOptions: Partial<InstanceOptions> = {}) {
    return IOClientFactory.createClient<Rewriter>(Rewriter, customContext, customOptions)
  }
``` 

In order to streamline the solution here I decided to use the HTTP client and send a POST request instead of a GET with data, such approach was able to fix the issue.

#### How should this be manually tested?
1. Clone the repo
2. Link the plugin using `vtex plugins link @vtex/cli-plugin-redirects`
3. When you run `vtex plugins list` a symlink should down in the plugins redirect
4. Run `vtex redirects export redirects.csv --verbose`


#### Screenshots or example usage

![image](https://github.com/user-attachments/assets/1ea6c53e-c5ff-451d-9fab-238e76752339)


#### Types of changes

#### Chores checklist

[XTNSNS-737]: https://vtex-dev.atlassian.net/browse/XTNSNS-737?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
